segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/ContainerTableExtensionImpl.java

```diff
@@ -320,9 +332,15 @@ protected int getMaxCompactionSize() {
     }
 
     private <T> CompletableFuture<Long> commit(Collection<T> toCommit, Function<Collection<T>, BufferView> serializer,
-                                               DirectSegmentAccess segment, Duration timeout) {
-        BufferView s = serializer.apply(toCommit);
-        return segment.append(s, null, timeout);
+                                               DirectSegmentAccess segment, long tableSegmentOffset, Duration timeout) {
+        if (tableSegmentOffset == NULL_TABLE_SEGMENT_OFFSET || tableSegmentOffset == segment.getInfo().getLength()) {
+            BufferView s = serializer.apply(toCommit);
+            return segment.append(s, null, timeout);
+        } else {
+            CompletableFuture<Long> future = new CompletableFuture<>();
```

Comment:
Invoke the appropriate `DirectSegmentAccess.append` method here instead of throwing this.